cli/tests/conftest.py

```diff
@@ -359,6 +359,12 @@ def _run_semgrep(
     return result
 
 
+@pytest.fixture()
```

Comment:
Can you explain (in comments in the code) this code? It's rather cryptic to me.
What's monkeypatch?

Reply:
Added a comment. `monkeypatch` allows you to set/get/del attributes on any imported class. It is often used in tests for usecases like this one, where the return value of a method needs to be overridden. 

In this particular case, monkeypatch is used to override the `Path.home` return value with the temporary directory created by the `tmp_path` fixture. When the `test_api` test runs and the settings file is read from `Path.home() / ".config/.semgrep/settings.yml"`, instead of checking `/home/runner/.config/.semgrep/settings.yml` it will check `/tmp/.../.config/.semgrep/settings.yml`